source/opt/dead_branch_elim_pass.cpp

```diff
@@ -250,7 +255,8 @@ bool DeadBranchElimPass::FixPhiNodesInLiveBlocks(
           modified = true;
           uint32_t continue_id = block.ContinueBlockIdIfAny();
           if (!backedge_added && continue_id != 0 &&
-              unreachable_continues.count(GetParentBlock(continue_id))) {
+              unreachable_continues.count(GetParentBlock(continue_id)) &&
+              operands.size() > 4) {
```

Comment:
I think this should be >=. In this case we haven't added a backedge (because the backedge block was the successor of an unreachable continue). So we have two valid incoming edges and would have a third from the unreachable continue.

Apparently no test covers this properly. Also please double check the subsequent code which special cases a two input phi node.

Reply:
In `operands`, the first two entries are the result id, and the result type.  If the first incoming egdse covers elements 3 and 4.  If there are move than 4, it is because there is a second incoming edge.

Do you want me to add a comment in the code?

Reply:
Ok, clearly forgot how my own code worked.